lnbits/decorators.py

```diff
@@ -102,16 +128,50 @@ async def __call__(self, request: Request) -> WalletTypeInfo:
             )
 
         request.scope["user_id"] = wallet.user
-        if self.expected_key_type is KeyType.admin and wallet.adminkey != key_value:
+        await _check_user_access(request, wallet.user)
+
+        key_type = await self._extract_key_type(key_value, wallet)
+        return WalletTypeInfo(key_type, wallet)
+
+
+class LightKeyChecker(BaseKeyChecker):
```

Comment:
i think you should just use 1 keychecker and add another argument and switch the crud getter on it. also cache all keychecks

am i missing something?